File: CivitasRuby/Sorpresa_PagarCobrar.rb
module Civitas
    class Sorpresa_PagarCobrar < Sorpresa
        public_class_method :new
        def initialize(valor,texto)
            super(texto)
            @valor=valor
        end

        def aplicarAJugador(actual, todos)

            if jugadorCorrecto(actual,todos)
                informe(actual, todos)
                todos[actual].modificarSaldo(@valor)
            end
        
        end

        def toString
            return @texto
        end
    end
end

File: CivitasJava/civitas/Sorpresa_PagarCobrar.java
package civitas;

import java.util.ArrayList;

class Sorpresa_PagarCobrar extends Sorpresa
{
    private int valor;
    
    Sorpresa_PagarCobrar(int valor, String texto)
    {
        super(texto);
        this.valor=valor;
    }

    @Override
    protected void aplicarAJugador(int actual, ArrayList<Jugador> todos)
    {
        if(super.jugadorCorrecto(actual, todos))
        {
            super.informe(actual, todos);
            todos.get(actual).modificarSaldo(valor);
        }
    }

    @Override
    public String toString()
    {
        return this.texto;
    }
}